### PR TITLE
fix: support sub-column for informix

### LINKED ISSUE #1832: [BUG] JSQLParser Version 4.6: Informix : Encountered unexpected token: "," ","
### Failing SQL Feature:
- can not parser the col[m, n] syntax

### SQL Example:
- Simplified Query Example, focusing on the failing feature
    ```sql
    select age, name, gender, citiy_code
    from user_info where city_code[1, 4] = '3301'
    ```

### Software Information:
- JSqlParser version 4.6
- Informix

### Error Info:
```java
net.sf.jsqlparser.JSQLParserException: net.sf.jsqlparser.parser.ParseException: Encountered unexpected token: "," ","
    at line 19, column 16.

Was expecting one of:

    "&"
    ":"
    "::"
    "<<"
    ">>"
    "COLLATE"
    "["
    "]"
    "^"
    "|"

```

### PR DESCRIPTION
**Pull request by SheldonKubor**
try to close #1832

### CODE REVIEW

**Comment by manticore-projects** [comment]
Greetings!

I need to ask: does this really belong to the `Column` rather then to an `ArrayExpression`?
We really need to avoid having 2 different interpretations of arrays.

File: src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt [hunk 1]
```diff
@@ -1806,17 +1806,26 @@ Column Column() #Column :
     List<String> data = new ArrayList<String>();
     ArrayConstructor arrayConstructor = null;
     Token tk = null;
+    Token subColumnFirst = null;
+    Token subColumnLast = null;
 }
 {
     data = RelObjectNameList()
+    ( LOOKAHEAD(5) "[" subColumnFirst=<S_LONG> "," subColumnLast=<S_LONG> "]" )?
```
Comment (manticore-projects) [inline src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt:1814]: is this really about `S_LONG` only? No variables or expressions here?

File: src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt [hunk 2]
```diff
@@ -1806,17 +1806,26 @@ Column Column() #Column :
...
     [ LOOKAHEAD(2) <K_COMMENT> tk=<S_CHAR_LITERAL> ]
     // @todo: we better should return a SEQUENCE instead of a COLUMN
     [ "." <K_NEXTVAL> { data.add("nextval"); } ]
 
-    [ LOOKAHEAD(2) arrayConstructor = ArrayConstructor(false) ]
+    [ LOOKAHEAD(5) arrayConstructor = ArrayConstructor(false) ]
```
Comment (manticore-projects) [inline src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt:1819]: I am not convinced that this is the right logical order or interpretation. How can we still have an `ArrayConstructor` after having this new `Column` format?

**Review by manticore-projects — CHANGES_REQUESTED** [review]
Please use `ArrayConstructor` instead. Those calls are 2 dimensional arrays as far as I understand it, aren't them?

**Comment by SheldonKubor** [comment]
> Greetings!
> 
> I need to ask: does this really belong to the `Column` rather then to an `ArrayExpression`? We really need to avoid having 2 different interpretations of arrays.

In my opinion,  for informix , there is a special sub-column syntax just like `col[m, n]`, just two sub-column, `ArrayExpression` can be lots of element

**Comment by manticore-projects** [comment]
but how are those different from Arrays?
I really want to avoid an overlapping implementation here for only one exotic RDBMS.

**Comment by SheldonKubor** [comment]
> Contributor

OK, there's not much difference, I will close this pr. Sorry for that. I thought too much. Thanks for your time!

**Comment by manticore-projects** [comment]
Sorry brother, shit can happen to the best.
Thank you big time for your kind help and support.

**Pull request closed without merging**